oauthdoctor/diag/config.go

```diff
@@ -187,7 +188,9 @@ func (c *ConfigFile) ReplaceConfigFromReader(key, value string, r io.Reader) str
 		langKey := c.GetConfigKeysInLang(key)
 
 		// Found the line with old config key and comment it out
-		if !strings.HasPrefix(trimmedLine, commentChar) && strings.Contains(trimmedLine, langKey) {
+		if c.Lang == "dotnet" && strings.Contains(trimmedLine, langKey) {
```

Comment:
If c.Lang != "dotnet" it will execute the else if where you do not check for c.Lang. So, the second and third conditions could conceivably test true with out c.Lang == "dotnet". Is that what you want?

Reply:
Thank you. That's not what I want. Will fix.